### PR TITLE
WT-7367 Do not remove unstable updates of an in-memory database btree page

### PR DESCRIPTION
**Pull request by kommiharibabu**
Remove the updates from the page only when both the start and stop time windows
are globally visible if the update has both time windows. Otherwise, the start time window.

Change the default value of rollback to stable to check for any unstable updates of a page
when the page doesn't have any aggregated time window information to avoid
rollback to stable skipping any pages with unstable updates.

### CODE REVIEW

File: src/reconcile/rec_visibility.c [hunk 1]
```diff
@@ -216,8 +216,11 @@ __rec_need_save_upd(
     if (F_ISSET(r, WT_REC_CHECKPOINT) && upd_select->upd == NULL)
         return (false);
 
-    return (!__wt_txn_tw_stop_visible_all(session, &upd_select->tw) &&
-      !__wt_txn_tw_start_visible_all(session, &upd_select->tw));
+    if (WT_TIME_WINDOW_HAS_STOP(&upd_select->tw))
```
Comment (quchenhao) [inline src/reconcile/rec_visibility.c:219]: Why we need this?

Comment (kommiharibabu) [reply src/reconcile/rec_visibility.c:219]: The older approach is if there is no stop timestamp it always returns false. The problem is when there is a stop timestamp, we should check both the start and stop should be globally visible, but the old check will pass if any one of them is globally visible. So I changed the check to check for both stop and start only if the stop is present.

Comment (quchenhao) [reply src/reconcile/rec_visibility.c:219]: Should we only need to check the stop timestamp to be globally visible if there is a stop timestamp?

Comment (quchenhao) [reply src/reconcile/rec_visibility.c:219]: If there is a stop timestamp and it is not globally visible, we still cannot discard the update chain.

Comment (kommiharibabu) [reply src/reconcile/rec_visibility.c:219]: Earlier by default RTS skips the pages that don't have the timestamp information, now with this error, I changed it to don't skip those pages, but it is better to fix the behaviour of reconciliation to not remove the unstable updates from the page.

Also, I changed the condition to check only to global visibility of the stop timestamp.

**Review by quchenhao — APPROVED** [review]
lgtm

File: test/suite/test_rollback_to_stable18.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        self.large_updates(uri, value_a, ds, nrows, self.prepare, 20)
+
+        # Perform several removes.
+        self.large_removes(uri, ds, nrows, self.prepare, 30)
+
+        # Verify data is visible and correct.
+        self.check(value_a, uri, nrows, 20)
+        self.check(None, uri, 0, 30)
+
+        evict_cursor = self.session.open_cursor(uri, None, "debug=(release_evict)")
```
Comment (raviprakashgiri29) [inline test/suite/test_rollback_to_stable18.py:89]: Can we add a comment here to be more readable "Configure the cursor to evict the page positioned on when the reset API is used" ?

**Review by raviprakashgiri29 — APPROVED** [review]
LGTM with some minor comments

File: test/suite/test_rollback_to_stable18.py (hunk 2 above)
Comment (kommiharibabu) [reply test/suite/test_rollback_to_stable18.py:90]: Added a comment to explain the debug option.

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Pull request merged**